.circleci/config.yml

```
@@ -93,6 +93,75 @@ commands:
             - bytecode-report-<< parameters.label >>-cli.txt
       - matrix_notify_failure_unless_pr
 
+  install_python3:
+    description: "Install python3 and given packages."
+    parameters:
+      packages:
+        description: "List of optional python packages to be installed (separated by space)."
+        type: string
+        default: ""
+    steps:
+      - run:
+          name: Install python and dependencies
+          command: |
+            sudo apt update
+            if ! python3 --version >/dev/null 2>&1
+            then
+              sudo apt install python3 --assume-yes --no-install-recommends
+            fi
+            sudo apt install python3-pip
```

Was unconditional installation causing a problem in some cases where Python is already installed? We generally don't mind getting newest Python so this would be fine too if it works:

```suggestion
            sudo apt install python3 python3-pip --assume-yes --no-install-recommends
```

---

No, I was just trying to avoid installing it if it was already in the image, since the command is used by different images.